File: WYNKWeb/src/app/Models/ViewModels/OfferViewM.ts
import { Offer } from '../Offer';
import { OfferTran } from '../OfferTran';



export class OfferViewM {

  Offer: Offer = new Offer();
  OfferTran: OfferTran = new OfferTran();

  getModel: Array<getModel> = [];
  ListItems: Array<ListItems> = [];
  ModelItems: Array<ModelItems> = [];
  ModelItems2: Array<ModelItems2> = [];
  ModelItems3: Array<ModelItems3> = [];
  oneplusOfferItems: Array<oneplusOfferItems> = [];
  freemodls: Array<freemodls> = [];
}


export class getModel {
  Text: string;
  Value: number;
}

export class ListItems {

  Models: string;
}

export class ModelItems {

  Models: string;
  LMID: number;
  LTID: number;
  
}

export class ModelItems2 {
  BrandID: number;
  BrandName: string;
  Models: string;
  LMID: number;
  LTID: number;
  OfferType: string;
}


export class ModelItems3 {
  BrandID: number;
  BrandName: string;
  Models: string;
  LMID: number;
  LTID: number;
  OfferType: string;
}



export class oneplusOfferItems {

  LMID: number;
  LTID: number;
  Models: string;
  OfferLMID: number;
  OfferLTID: number;
  OfferType: string;
  oneplusModel: string;
}

export class freemodls{

 
  Value: number;
  Values: number;

}

File: WYNK.Data/Model/ViewModel/OfferViewM.cs
using System;
using System.Collections.Generic;
using System.Text;
namespace WYNK.Data.Model.ViewModel
{
    public class OfferViewM
    {
        public Offer Offer { get; set; }
        public OfferTran OfferTran { get; set; }

        public ICollection<getModel> getModel { get; set; }
        public ICollection<oneplusmodels> oneplusmodels { get; set; }
        public ICollection<ModelItems> ModelItems { get; set; }
        public ICollection<ModelItems2> ModelItems2 { get; set; }
        public ICollection<oneplusOfferItems> oneplusOfferItems { get; set; }

    }

    public class getModel
    {
        public string Models { get; set; }
        public string LMID { get; set; }
        public int LTID { get; set; }
        public int BrandID { get; set; }
        public string BrandName { get; set; }
    }

    public class ModelItems2
    {

        public string Models { get; set; }
        public int LTID { get; set; }
        public int LMID { get; set; }
        public string OfferType { get; set; }
    }

    public class ModelItems
    {

        public string Models { get; set; }
        public int LTID { get; set; }
        public int LMID { get; set; }
    }



    public class oneplusmodels
    {
        public int OP_LTID { get; set; }
        public string OP_LMID { get; set; }
        public string OP_Model { get; set; }
    }



    public class oneplusOfferItems
    {

        public int LMID { get; set; }
        public int LTID { get; set; }
        public string Models { get; set; }
        public int OfferLMID { get; set; }
        public int OfferLTID { get; set; }
        public string OfferType { get; set; }
        public string oneplusModel { get; set; }

    }
}
